src/Abstractions/CharBoard.cs

```diff
@@ -72,4 +76,142 @@ public CharBoard Set(int r, int c, char value)
             return new CharBoard(Size, newCells);
         }
     }
+    
+    public record DiceBoard
+    {
+        private static readonly ConcurrentDictionary<int, DiceBoard> EmptyCache = new();
+        public static DiceBoard Empty(int size) => EmptyCache.GetOrAdd(size, size1 => new DiceBoard(size1));
+
+        public int Size { get; }
+        public ImmutableDictionary<int, Cell> Cells { get; }
+        
+        public struct Cell
+        {
+            public string Background;
+            public string[] Colors;
+            public double[] Opacities;
+
+            public Cell(string background, string[] colors, double[] opacities)
+            {
+                var defaultBackground = GetColor(DiceBoard.Colors.Gold);
+                var defaultColors = new string[] {GetColor(DiceBoard.Colors.Blue),
+                    GetColor(DiceBoard.Colors.Green),
+                    GetColor(DiceBoard.Colors.Red),
+                    GetColor(DiceBoard.Colors.Yellow), };
+                var defaultOpacities = new double[] {
+                    GetOpacity(Opacity.Invisible), GetOpacity(Opacity.Invisible), GetOpacity(Opacity.Invisible),
+                    GetOpacity(Opacity.Invisible),
+                };
+                background ??= defaultBackground;
+                colors ??= defaultColors;
+                opacities ??= defaultOpacities;
+                Background = background;
+                Colors = colors;
+                Opacities = opacities;
+            }
+        }
+        
+        public enum Colors
+        {
+            Blue,
```

Comment:
I would rather use CellStates vs Colors. Color is meaningless, but cell state - e.g. "BoostCell" - is meaninful.